### PR TITLE
Feature/add location and store

### PR DESCRIPTION
**Pull request by SvitlanaVorobets**
Implemented:
- store for getting and adding locations
- filtering locations
- adding locations

### CODE REVIEW

File: src/app/ubs/ubs-admin/components/ubs-admin-tariffs/ubs-admin-tariffs-location-dashboard.component.ts [hunk 1]
```diff
@@ -35,19 +63,134 @@ export class UbsAdminTariffsLocationDashboardComponent implements OnInit, OnDest
...
+      this.selectCity(event);
+    }
+    this.positionsFilter();
+    this.city.setValue('');
+    requestAnimationFrame(() => {
+      trigger.openPanel();
+    });
+  }
+
+  selectCity(event) {
```
Comment (hnativlyubomyr) [inline src/app/ubs/ubs-admin/components/ubs-admin-tariffs/ubs-admin-tariffs-location-dashboard.component.ts:143]: add event type

File: src/app/ubs/ubs-admin/components/ubs-admin-tariffs/ubs-admin-tariffs-location-dashboard.component.ts [hunk 2]
```diff
@@ -35,19 +63,134 @@ export class UbsAdminTariffsLocationDashboardComponent implements OnInit, OnDest
...
+    const newValue = event.option.viewValue;
+    if (this.checkedCities.includes(newValue)) {
+      this.checkedCities = [...this.checkedCities.filter((item) => item !== newValue)];
+    } else {
+      this.checkedCities.push(event.option.viewValue);
+    }
+  }
+
+  isChecked() {
```
Comment (hnativlyubomyr) [inline src/app/ubs/ubs-admin/components/ubs-admin-tariffs/ubs-admin-tariffs-location-dashboard.component.ts:152]: add return type (boolean)

File: src/app/ubs/ubs-admin/components/ubs-admin-tariffs/ubs-admin-tariffs-location-dashboard.component.ts [hunk 3]
```diff
@@ -35,19 +63,134 @@ export class UbsAdminTariffsLocationDashboardComponent implements OnInit, OnDest
...
+    return this.checkedCities.length === this.cities.length;
+  }
+
+  isEmpty() {
```
Comment (hnativlyubomyr) [inline src/app/ubs/ubs-admin/components/ubs-admin-tariffs/ubs-admin-tariffs-location-dashboard.component.ts:156]: the same

File: src/app/ubs/ubs-admin/components/ubs-admin-tariffs/ubs-admin-tariffs-location-dashboard.component.ts [hunk 4]
```diff
@@ -35,19 +63,134 @@ export class UbsAdminTariffsLocationDashboardComponent implements OnInit, OnDest
...
+    return this.filteredLocations.length === 0;
+  }
+
+  positionsFilter() {
```
Comment (hnativlyubomyr) [inline src/app/ubs/ubs-admin/components/ubs-admin-tariffs/ubs-admin-tariffs-location-dashboard.component.ts:160]: add return type (void)

File: src/app/ubs/ubs-admin/components/ubs-admin-tariffs/ubs-admin-tariffs-location-dashboard.component.ts [hunk 5]
```diff
@@ -35,19 +63,134 @@ export class UbsAdminTariffsLocationDashboardComponent implements OnInit, OnDest
...
+    if (this.checkedCities.length !== 0) {
+      this.filteredLocations = this.onPositionSelected();
+    } else {
+      this.filteredLocations = this.locations;
+    }
+  }
+
+  onPositionSelected() {
```
Comment (hnativlyubomyr) [inline src/app/ubs/ubs-admin/components/ubs-admin-tariffs/ubs-admin-tariffs-location-dashboard.component.ts:168]: add return type

File: src/app/ubs/ubs-admin/components/ubs-admin-tariffs/ubs-admin-tariffs-location-dashboard.component.ts [hunk 6]
```diff
@@ -35,19 +63,134 @@ export class UbsAdminTariffsLocationDashboardComponent implements OnInit, OnDest
...
+      const res = user.locationsDto.map((it) =>
+        it.locationTranslationDtoList.filter((item) => this.checkedCities.includes(item.locationName))
+      );
+      if (!isEmpty(res)) {
+        return true;
+      }
+    });
+  }
+
+  openAuto(event, trigger: MatAutocompleteTrigger) {
```
Comment (hnativlyubomyr) [inline src/app/ubs/ubs-admin/components/ubs-admin-tariffs/ubs-admin-tariffs-location-dashboard.component.ts:180]: add event and return type

File: src/app/ubs/ubs-admin/components/ubs-admin-tariffs/ubs-admin-tariffs-location-dashboard.component.ts [hunk 7]
```diff
@@ -35,19 +63,134 @@ export class UbsAdminTariffsLocationDashboardComponent implements OnInit, OnDest
...
+    event.stopPropagation();
+    trigger.openPanel();
+  }
+
+  toggleSelectAll() {
```
Comment (hnativlyubomyr) [inline src/app/ubs/ubs-admin/components/ubs-admin-tariffs/ubs-admin-tariffs-location-dashboard.component.ts:185]: the same

File: src/app/ubs/ubs-admin/components/ubs-admin-tariffs/ubs-admin-tariffs-location-dashboard.component.ts [hunk 8]
```diff
@@ -64,13 +207,16 @@ export class UbsAdminTariffsLocationDashboardComponent implements OnInit, OnDest
   }
 
   getLocations() {
```
Comment (hnativlyubomyr) [inline src/app/ubs/ubs-admin/components/ubs-admin-tariffs/ubs-admin-tariffs-location-dashboard.component.ts:209]: the same

File: src/app/ubs/ubs-admin/components/ubs-admin-tariffs/ubs-admin-tariffs-location-pop-up/ubs-admin-tariffs-location-pop-up.component.ts [hunk 9]
```diff
@@ -123,42 +138,66 @@ export class UbsAdminTariffsLocationPopUpComponent implements OnInit, AfterViewC
...
+
       this.locationForm.get('items')[key][i][key].location.setValue(locationName);
       this.translate(locationName, this.locationForm.get('items')[key][i][key].englishLocation);
-      this.locationForm.get('items')[key][i][key].englishLocation.setValue(this.translatedText);
+      this.locationForm.get('items')[key][i][key].latitude.setValue(latitude);
+      this.locationForm.get('items')[key][i][key].longitude.setValue(longitude);
     });
   }
 
   addLocation() {
```
Comment (hnativlyubomyr) [inline src/app/ubs/ubs-admin/components/ubs-admin-tariffs/ubs-admin-tariffs-location-pop-up/ubs-admin-tariffs-location-pop-up.component.ts:178]: the same

**Review by NewArtMask — APPROVED** [review]

**Review by NazarDobriy — APPROVED** [review]

**Review by TarasKuzma1999 — APPROVED** [review]

**Pull request merged**